deploy/kubernetes/base/controller-deployment.yaml

```diff
@@ -1,34 +1,37 @@
 ---
-kind: Deployment
+# Source: aws-fsx-csi-driver/templates/controller-deployment.yaml
 apiVersion: apps/v1
+kind: Deployment
 metadata:
   name: fsx-csi-controller
-  namespace: kube-system
+  labels:
+    app.kubernetes.io/name: aws-fsx-csi-driver
 spec:
   replicas: 2
   selector:
     matchLabels:
       app: fsx-csi-controller
+      app.kubernetes.io/name: aws-fsx-csi-driver
```

Comment:
I don't know if this is a concern but this will need intervention on a new apply since matchLabels are immutable.

Reply:
This may still be the case since the deployment name isn't changing.  Did you test the upgrade with kustomize?